native/src/seal/polyarray.h

```diff
@@ -205,20 +241,15 @@ namespace seal
                 // PolynomialArray without knowing the modulus yet (it
                 // gets passed in later)
                 void set_modulus(const std::vector<Modulus> &coeff_modulus){
+                    coeff_modulus_ = coeff_modulus;
                     coeff_modulus_size_ = coeff_modulus.size();
 
-                    // Other classes do a check that the modulus set is correct;
-                    // the assumption here is that we are creating an instance
-                    // of this class from an existing (verified) modulus set.
-                    coeff_modulus_ = allocate<Modulus>(coeff_modulus_size_, pool_);
-                    copy_n(coeff_modulus.cbegin(), coeff_modulus_size_, coeff_modulus_.get());
-
                     rnsbase_ = allocate<RNSBase>(pool_, coeff_modulus, pool_);
                 }
 
                 MemoryPoolHandle pool_;
                 Pointer<uint64_t> data_;
-                Pointer<Modulus> coeff_modulus_;
+                std::vector<Modulus> coeff_modulus_;
```

Comment:
I hate when people create their own memory managers, but since we already have one, is there a reason to not use it?

Reply:
I had a hard time using the value when instantiating the copy constructor and the coefficient modulus is passed around as a vector everywhere else, so it seemed prudent to do so here.